test/should-fail.test.tsx

```diff
@@ -218,14 +218,6 @@ glamorous('div', {
   withProps: ''
 })
 
```

Comment:
I'm assuming this was removed by accident as it looks like it should fail and appears unrelated to the changes. Would you mind re adding if that's the case thanks.

Reply:
This is removed on purpose, as this relaxation is the side-effect of the proposed change.

The new indexed type allows the value to be boolean, which makes the `primaryColor: props.visible` be valid.

Of course, it’s silly that we have this effect, but as noted in the other comment, it’d be awesome to not even have `children` in the `CSSProperties`. Until then this is a workaround.

The PR is effectively relaxing the type-checking for this instance, meaning fewer errors will be caught by TypeScript compiler. However, requirements that are too lax are much better than requirements that are too strict. In the latter case the developers fall into habit into silencing errors (with `as any`), defeating the entire purpose of typechecking in the first place.